tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1019,6 +1047,19 @@ def _train_single_full(self, X_train, y_train, model: AbstractModel, X_unlabeled
             time_limit=time_limit,
             verbosity=self.verbosity,
         )
+        if self.weight_column is not None:
+            X_train, w_train = extract_column(X_train, self.weight_column)
+            if w_train is not None:  # may be None for ensemble
+                weights_normalized = w_train.values/w_train.sum() * len(w_train)  # normalization can affect gradient algorithms like boosting
```

Comment:
Why here? Wouldn't it make more sense to do this in the actual model code? Such as in the generic `model.fit` function that is applied before going to a model's `model._fit` implementation?

Reply:
Added TODO to consider moving this into AbstractModel.fit().
However I'm not sure why it's preferred there, this weight normalization is not really part of the model fitting. Also it would look a bit odd to add this in current AbstractModel.fit() which does not even list `X_train` as an argument right now.